internal/http/http.go

```diff
@@ -28,3 +32,41 @@ func SetCacheHeaders(w http.ResponseWriter, maxAge time.Duration, timeUntilExpir
 	w.Header().Set("Cache-Control", "public, max-age="+maxAge.String())
 	w.Header().Set("Expires", time.Now().Add(timeUntilExpiry).Format(time.RFC1123))
 }
+
+func LimitRead(r io.Reader) ([]byte, error) {
+	lr := io.LimitReader(r, maxBytes)
+
+	// Read as far as we are allowed to
+	bodyBytes, err := io.ReadAll(lr)
+	if err != nil {
+		return nil, Error(
+			fmt.Errorf(
+				"failed to read from reader: %w", err,
+			),
+			http.StatusBadRequest,
+		)
```

Comment:
This is an unanticipated problem, meaning an internal server error, which [doesn't need to be wrapped](https://github.com/akuity/kargo/pull/4000/files#r2082518344).

```suggestion
		return nil, fmt.Errorf("failed to read from reader: %w", err)
```